### PR TITLE
feat/CART-CPC-1255 change the List of product ids to a list of product objects

### PR DESCRIPTION
**Pull request by Hares-2088**
**JIRA:** https://champlainsaintlambert.atlassian.net/browse/CPC-1284
## Context:
- updated the docker-compose_no_FE.yml to integrate the cart-service
- finished changing everything to use CartProduct instead of a list of productIds
- updated the front end to work with the new version of the cart service
- added a go back button in the user cart page
## Does this PR change the .vscode folder in petclinic-frontend?:
no, it doesn't
<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
- I changed the entity so therefore almost every other class so that the requests work.
- Someone commented on some lines of codes in the front end so I put them back
- I commented on many tests for now. I'm going to write them again. i did it so that my teammates could work on their part

### CODE REVIEW

File: cart-service/src/test/java/com/petclinic/cartsservice/businesslayer/CartServiceUnitTest.java
Comment (AlexCristea04) [inline cart-service/src/test/java/com/petclinic/cartsservice/businesslayer/CartServiceUnitTest.java:1]: A lot of these tests shouldn't be commented out. Please revise.

**Review by AlexCristea04 — CHANGES_REQUESTED** [review]
Please revise the commented tests in CartServiceUnitTest

File: cart-service/src/test/java/com/petclinic/cartsservice/businesslayer/CartServiceUnitTest.java
Comment (Hares-2088) [reply cart-service/src/test/java/com/petclinic/cartsservice/businesslayer/CartServiceUnitTest.java:1]: I forgot to put them back. thank you for bringing it to my attention. it is fixed

**Review by AlexCristea04 — DISMISSED** [review]
Works fine on my machine and changes were made according to my requests!

**Review by NtsoaElie — APPROVED** [review]
Implements everything as we discussed during our Sprint Planning and brainstorming. Good job!

**Review by AlexCristea04 — APPROVED** [review]
Minor changes since my last approval. Everything looks good

**Pull request merged**